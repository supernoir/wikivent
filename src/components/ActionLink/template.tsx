import React from "react"
import { StyledActionLink, StyledLabel } from "./styles"
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faTrash, faCheckCircle, faPencilAlt } from '@fortawesome/free-solid-svg-icons'

export interface ActionLinkProps {
  id: string,
  type: ActionLinkType,
  onClick: (evt: React.MouseEvent) => void
}

export enum ActionLinkTypes {
  edit = "edit",
  verify = "verify",
  delete = "delete"
}

export type ActionLinkType = ActionLinkTypes.edit | ActionLinkTypes.verify | ActionLinkTypes.delete

export const ActionLink: React.FC<ActionLinkProps> = ({ id, type, onClick }) => {
  switch (type) {
    case ActionLinkTypes.delete:
      return <StyledActionLink onClick={onClick} id={id}>
        <FontAwesomeIcon icon={faTrash} /><StyledLabel>{"Delete"}</StyledLabel>
      </StyledActionLink>
    case ActionLinkTypes.verify:
      return <StyledActionLink onClick={onClick} id={id}>
        <FontAwesomeIcon icon={faCheckCircle} /><StyledLabel>{"Verify"}</StyledLabel>
      </StyledActionLink>
    default:
      return <StyledActionLink onClick={onClick} id={id}>
        <FontAwesomeIcon icon={faPencilAlt} /><StyledLabel>{"Edit"}</StyledLabel>
      </StyledActionLink>
  }
}