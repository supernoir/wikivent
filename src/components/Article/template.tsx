import React from "react"
import { Paper } from "../Paper"
import { Section } from "../Section"
import { Paragraph } from "grommet"

export interface ArticleProps {
  title: string,
  children: React.ReactNode
}

export const Article: React.FC<ArticleProps> = ({ title, children }) => {
  return (
    <Paper>
      <Section>
        <h1>{title}</h1>
        <Paragraph fill>
          {children}
        </Paragraph>
      </Section>
    </Paper>
  )
}
