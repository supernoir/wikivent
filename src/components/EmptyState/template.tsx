import React from "react"

export const EmptyState: React.FC<{ message: string }> = ({ message }) => {
  return <div>
    <h3>{message}</h3>
  </div>
}