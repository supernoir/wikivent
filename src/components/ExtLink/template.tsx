import React from "react"
import { ExternalLink } from "./styles"
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faExternalLinkAlt } from '@fortawesome/free-solid-svg-icons'

export interface ExtLinkProps {
  link: string,
  label: string
}

export const ExtLink: React.FC<ExtLinkProps> = ({ link, label }) => {
  return (
    <ExternalLink href={link}><FontAwesomeIcon icon={faExternalLinkAlt} /> {label}</ExternalLink>
  )
}