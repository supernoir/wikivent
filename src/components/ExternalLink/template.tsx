import React from "react"
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faExternalLinkAlt } from '@fortawesome/free-solid-svg-icons'
import { Anchor } from "./styles"

export interface ExtLinkProps {
  link: string,
  label: string
}

export const ExternalLink: React.FC<ExtLinkProps> = ({ link, label }) => {
  return (
    <Anchor href={link}><FontAwesomeIcon icon={faExternalLinkAlt} /> {label}</Anchor>
  )
}