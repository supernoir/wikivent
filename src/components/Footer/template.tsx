import React from 'react'
import { StyledFooter } from './styles'

export const Footer: React.FC = () => {
  return (<StyledFooter>
    <span>&copy; Wikivent 2020</span>
  </StyledFooter>)
}