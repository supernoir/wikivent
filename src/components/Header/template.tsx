import React from 'react'
import { Name, Highlight, Logo, StyledLogo, StyledHeader } from './styles'
import { Navigation } from '../Navigation/template'

export const Header: React.FC = () => {
  return (
    <StyledHeader>
      <Logo><StyledLogo /><Name>WIKI<Highlight>Vent</Highlight></Name></Logo>
      <Navigation />
    </StyledHeader>
  )
}
