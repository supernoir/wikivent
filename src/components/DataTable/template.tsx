import React from 'react'
import {
  Table,
  TableHead,
  TableBody,
  TableHeadRow,
  TableCell,
  FeatureList,
  FeatureListItem,
  DataTableWrapper,
  ActionLinkList
} from './styles'

import { Loader } from '../Loader/template'
import { ExternalLink } from '../ExternalLink'
import { TextLink } from '../TextLink/styles'
import { EmptyState } from '../EmptyState/template'
import { DataContextType, DataContext } from '../../types/inventory/VentilatorTypes'
import { ActionLinkTypes, ActionLink } from '../ActionLink/template'
import axios from "axios"
import { toast } from "react-toastify"


const tablehead = [
  "Type",
  "Make",
  "Model",
  "Features",
  "Link"
]

const tableheadSubmitted = [
  "Type",
  "Make",
  "Model",
  "Features",
  "Link",
  "Actions"
]

export interface DataTableInterface {
  data: any,
  context: DataContextType,
  isLoading: boolean
}

export const DataTable: React.FC<DataTableInterface> = ({ data, context, isLoading }) => {

  const handleEdit = (evt: React.MouseEvent) => {
    evt.preventDefault()
    console.log("Edit!")
  }

  const handleVerify = (evt: React.MouseEvent) => {
    evt.preventDefault()
    console.log("Verify!")
  }

  const handleDelete = (evt: React.MouseEvent) => {
    evt.preventDefault()
    const id = evt.currentTarget.id

    axios.post(
      `http://localhost:8081/submitted/delete/${id}`,
    ).then(result => {
      if (result.status === 200) {
        toast.success("Ventilator submission deleted!")
      }
    })
      .catch(err => toast.error(err.message));

  }

  return (
    <DataTableWrapper>
      {
        isLoading
          ? <Loader />
          : <Table>
            <TableHead>
              <TableHeadRow>
                {context === DataContext.submitted
                  ? tableheadSubmitted.map((head) => <TableCell key={head}>{head}</TableCell>)
                  : tablehead.map((head) => <TableCell key={head}>{head}</TableCell>)
                }
              </TableHeadRow>
            </TableHead>
            <TableBody>
              {data && data.length > 0
                ? data.map((item: any) => {
                  return <tr key={item.id}>
                    <TableCell>{item.type}</TableCell>
                    <TableCell>{item.make}</TableCell>
                    <TableCell><TextLink to={`/detail/${item.id}`}>{item.model}</TextLink></TableCell>
                    <TableCell>
                      <FeatureList>{item && item.features && item.features.length > 0 && item.features.map((feature: any) => <FeatureListItem key={feature}>{feature}</FeatureListItem>)}</FeatureList>
                    </TableCell>
                    <TableCell>
                      <ExternalLink link={item.link} label={"Product page"} />
                    </TableCell>
                    {context === DataContext.submitted && <TableCell>
                      <ActionLinkList>
                        <ActionLink type={ActionLinkTypes.edit} onClick={handleEdit} id={item.id} />
                        <ActionLink type={ActionLinkTypes.verify} onClick={handleVerify} id={item.id} />
                        <ActionLink type={ActionLinkTypes.delete} onClick={handleDelete} id={item.id} />
                      </ActionLinkList>
                    </TableCell>
                    }

                  </tr>
                })
                : <EmptyState message={"No entries found"} />
              }
            </TableBody>
          </Table>
      }

    </DataTableWrapper>
  )
}
