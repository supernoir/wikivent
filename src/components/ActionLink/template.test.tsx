import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'
import { ActionLink, ActionLinkTypes } from './template'

const onClickMock = jest.fn()

test('Renders ActionLink with type delete', async () => {
  const container = render(<ActionLink type={ActionLinkTypes.delete} id={"test"} onClick={onClickMock} />)

  fireEvent.click(screen.getByText('Delete'))
  expect(onClickMock).toHaveBeenCalled()
  expect(container).toMatchSnapshot()
})

test('Renders ActionLink with type edit', async () => {
  const container = render(<ActionLink type={ActionLinkTypes.edit} id={"test"} onClick={onClickMock} />)

  fireEvent.click(screen.getByText('Edit'))
  expect(onClickMock).toHaveBeenCalled()
  expect(container).toMatchSnapshot()
})

test('Renders ActionLink with type verify', async () => {
  const container = render(<ActionLink type={ActionLinkTypes.verify} id={"test"} onClick={onClickMock} />)

  fireEvent.click(screen.getByText('Verify'))
  expect(onClickMock).toHaveBeenCalled()
  expect(container).toMatchSnapshot()
})