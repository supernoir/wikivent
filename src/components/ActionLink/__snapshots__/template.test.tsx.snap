// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders ActionLink with type delete 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="sc-AxjAm kRnQUN"
        id="test"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-trash fa-w-14 "
          data-icon="trash"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
            fill="currentColor"
          />
        </svg>
        <span
          class="sc-AxirZ fbAxwg"
        >
          Delete
        </span>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="sc-AxjAm kRnQUN"
      id="test"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-trash fa-w-14 "
        data-icon="trash"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
          fill="currentColor"
        />
      </svg>
      <span
        class="sc-AxirZ fbAxwg"
      >
        Delete
      </span>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Renders ActionLink with type edit 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="sc-AxjAm kRnQUN"
        id="test"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-pencil-alt fa-w-16 "
          data-icon="pencil-alt"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"
            fill="currentColor"
          />
        </svg>
        <span
          class="sc-AxirZ fbAxwg"
        >
          Edit
        </span>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="sc-AxjAm kRnQUN"
      id="test"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-pencil-alt fa-w-16 "
        data-icon="pencil-alt"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"
          fill="currentColor"
        />
      </svg>
      <span
        class="sc-AxirZ fbAxwg"
      >
        Edit
      </span>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Renders ActionLink with type verify 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="sc-AxjAm kRnQUN"
        id="test"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-check-circle fa-w-16 "
          data-icon="check-circle"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
            fill="currentColor"
          />
        </svg>
        <span
          class="sc-AxirZ fbAxwg"
        >
          Verify
        </span>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="sc-AxjAm kRnQUN"
      id="test"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-check-circle fa-w-16 "
        data-icon="check-circle"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
          fill="currentColor"
        />
      </svg>
      <span
        class="sc-AxirZ fbAxwg"
      >
        Verify
      </span>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`loads and displays greeting 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="sc-AxjAm kRnQUN"
        id="test"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-trash fa-w-14 "
          data-icon="trash"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
            fill="currentColor"
          />
        </svg>
        <span
          class="sc-AxirZ fbAxwg"
        >
          Delete
        </span>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="sc-AxjAm kRnQUN"
      id="test"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-trash fa-w-14 "
        data-icon="trash"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
          fill="currentColor"
        />
      </svg>
      <span
        class="sc-AxirZ fbAxwg"
      >
        Delete
      </span>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
